src/main/scala/org/allenai/scholar/metrics/Example.scala

```diff
@@ -0,0 +1,48 @@
+package org.allenai.scholar.metrics
+
+trait Example[+T] {
+  val trueLabels: Iterable[T]
+  val predictedLabels: Iterable[T]
```

Comment:
I'm a bit uncomfortable with `Iterable` since it can be non-strict and may get realized multiple times. An example is `scala.io.Source.fromFile(bigFile).getLines.toIterable`. If we call `size` on this `Iterable` twice, we'll read `bigFile` twice.